services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/domain/Hanke.kt

```diff
@@ -56,5 +64,38 @@ data class Hanke(
     var polyHaitta: Haitta13? = null
     var tarinaHaitta: Haitta13? = null
 
+    /** Note: this can be null for two reasons; the field wasn't requested for, or there are no geometries for the hanke.
+     * See 'tilaOnGeometrioita' field.
+     */
     var geometriat: HankeGeometriat? = null
+
+    // --------------- State flags -------------------
+    var tilat: HankeTilat = HankeTilat()
+
+    fun updateStateFlags() {
+        updateStateFlagOnKaikkiPakollisetLuontiTiedot()
+        updateStateFlagTiedotLiikHaittaIndeksille()
+    }
+
+    fun updateStateFlagOnKaikkiPakollisetLuontiTiedot() {
+        // All mandatory fields have been given... (though their validity should be checked elsewhere)
+        //  and saveType is submit, not just draft?
+        tilat.onKaikkiPakollisetLuontiTiedot = !nimi.isNullOrBlank()
+                && !kuvaus.isNullOrBlank()
+                && (alkuPvm != null) && (loppuPvm != null)
+                && (vaihe != null) && (vaihe != Vaihe.SUUNNITTELU || suunnitteluVaihe != null)
+                && !tyomaaKatuosoite.isNullOrBlank()
```

Comment:
Have you checked all these from Pihla? I don't think this is necessary for doing liikennehaittaindeksi

Reply:
Pihla was uncertain about that field.  I'll comment it for now..